### ISSUE TITLE
Texts missing spaces when using a screen reader

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#517]
**Describe the bug**
These texts are missing spaces when using a screen reader:

Parent menu item + description
Continue reading + post title
Edit link + post title

Can also be improved:
Previous post  + post title, Next post + post title
has no spacing after the colon, reads like: Previous post:title

**Additional context**
Tested with NVDA on windows 10
